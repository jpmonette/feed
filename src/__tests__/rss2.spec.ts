import { sampleFeed } from "./setup";

describe("rss 2.0", () => {
  it("should generate a valid feed", () => {
    let actual = sampleFeed.rss2();
    expect(actual).toMatchSnapshot();
  });
});
