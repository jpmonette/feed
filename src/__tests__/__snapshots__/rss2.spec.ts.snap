// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rss 2.0 should generate a valid feed 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"utf-8\\"?>
<rss version=\\"2.0\\" xmlns:content=\\"http://purl.org/rss/1.0/modules/content/\\" xmlns:atom=\\"http://www.w3.org/2005/Atom\\">
    <channel>
        <title>Feed Title</title>
        <link>http://example.com/</link>
        <description>This is my personnal feed!</description>
        <lastBuildDate>Sat, 13 Jul 2013 23:00:00 GMT</lastBuildDate>
        <docs>http://blogs.law.harvard.edu/tech/rss</docs>
        <generator>https://github.com/jpmonette/feed</generator>
        <language>en</language>
        <image>
            <title>Feed Title</title>
            <url>http://example.com/image.png</url>
            <link>http://example.com/</link>
        </image>
        <copyright>All rights reserved 2013, John Doe</copyright>
        <category>Technology</category>
        <atom:link href=\\"http://example.com/sampleFeed.rss\\" rel=\\"self\\" type=\\"application/rss+xml\\"/>
        <item>
            <title><![CDATA[Hello World]]></title>
            <link>https://example.com/hello-world</link>
            <guid>https://example.com/hello-world</guid>
            <pubDate>Sat, 13 Jul 2013 23:00:00 GMT</pubDate>
            <description><![CDATA[This is an article about Hello World.]]></description>
            <content:encoded><![CDATA[Content of my item]]></content:encoded>
            <author>janedoe@example.com (Jane Doe)</author>
            <author>joesmith@example.com (Joe Smith)</author>
            <enclosure url=\\"https://example.com/hello-world.jpg\\"/>
        </item>
    </channel>
</rss>"
`;
